app/src/main/java/info/nightscout/androidaps/plugins/general/wear/wearintegration/DataLayerListenerServiceMobile.kt

```diff
@@ -115,12 +114,37 @@ class DataLayerListenerServiceMobile : WearableListenerService() {
 
         if (wearPlugin.isEnabled()) {
             when (messageEvent.path) {
-                rxPath -> {
-                    aapsLogger.debug(LTag.WEAR, "onMessageReceived: ${String(messageEvent.data)}")
+                rxPath        -> {
+                    aapsLogger.debug(LTag.WEAR, "onMessageReceived rxPath: ${String(messageEvent.data)}")
                     val command = EventData.deserialize(String(messageEvent.data))
                     rxBus.send(command.also { it.sourceNodeId = messageEvent.sourceNodeId })
                 }
+
+                exceptionPath -> logWearException(messageEvent)
+            }
+        }
+    }
+
+    private fun logWearException(messageEvent: MessageEvent) {
```

Comment:
I'd move this to FabricPrivacy and do not expose `instance()`

Reply:
When moving the whole function, it feels a bit dirty to pass MessageEvent to FabricPrivacy... not separation of concern?

Reply:
you can decompile it here and then pass to FabricPrivacy as Exception